utils/generate_invite_url.py

```diff
@@ -1,128 +1,44 @@
-            )
-
-        discord_guild_id: str | None = parsed_args.discord_guild_id or None
-        if not discord_guild_id:
-            import dotenv
-
-            dotenv.load_dotenv()
-            discord_guild_id = os.getenv("DISCORD_GUILD_ID")
-
-            if not discord_guild_id:
-                cls._function_subparsers[parser].error(
-                    "discord_guild_id must be provided as an argument "
-                    "to the generate_invite_url utility function "
-                    "or otherwise set the DISCORD_GUILD_ID environment variable"
-                )
-
-        if not discord_guild_id or not re.match(r"\A\d{17,20}\Z", discord_guild_id):
-            cls._function_subparsers[parser].error(
-                "discord_guild_id must be a valid Discord guild ID (see https://docs.pycord.dev/en/stable/api/abcs.html#discord.abc.Snowflake.id)"
-            )
-
-        sys.stdout.write(f"""{cls.generate_invite_url(
-            parsed_args.discord_bot_application_id,
-            int(discord_guild_id)
-        )}\n""")
-        return 0
-
-    # noinspection PyShadowingNames
-    @staticmethod
-    def generate_invite_url(discord_bot_application_id: str, discord_guild_id: int) -> str:
-        """
-        Generate the correct OAuth invite URL for the bot.
-
-        This invite URL directs to the given Discord guild and requests only the permissions
-        required for the bot to run.
-        """
-        return discord.utils.oauth_url(
-            client_id=discord_bot_application_id,
-            permissions=discord.Permissions(
-                manage_roles=True,
-                read_messages=True,
-                send_messages=True,
-                manage_messages=True,
-                embed_links=True,
-                read_message_history=True,
-                mention_everyone=True,
-                add_reactions=True,
-                use_slash_commands=True,
-                kick_members=True,
-                ban_members=True,
-                manage_channels=True,
-                view_audit_log=True
-            ),
-            guild=discord.Object(id=discord_guild_id),
-            scopes=("bot", "applications.commands"),
-            disable_guild_select=True
-        )
+__all__: Sequence[str] = ("generate_invite_url",)
+
+def generate_invite_url(discord_bot_application_id: str, discord_guild_id: str) -> str:
+    """Execute the logic that this util function provides."""
+    discord_bot_application_id = str(discord_bot_application_id)
+    discord_guild_id = str(discord_guild_id)
+
+    if not discord_guild_id:
+        err = "discord_guild_id must be set in the DISCORD_GUILD_ID environment variable"
+        raise ValueError(err)
+
+    if not discord_guild_id or not re.match(r"\A\d{17,20}\Z", discord_guild_id):
```

Comment:
This check is not necessary if the argument is guaranteed to be an integer.

Reply:
Wouldn't this check be needed to check that the integer passed for discord_guild_id matches the desired format of 17 to 20 digits? I would understand removing the "not discord_guild_id" check of it.

Reply:
The check can be scrapped entirely because the value is validated within settings.py. The discord_guild_id is guarunteed to be valid at this point because the check occurred already. Take a look at settings.py to see the full set of validation checks that occur for all the environment variables